### PR TITLE
Try aligning tree view drop handling

### PR DESCRIPTION
**Pull request by mjbvz**
Tree views provided the first public drag and drop api. We've since also adopted drag and drop for dropping into editors as well, which resulted in some duplicated code between it and the tree view implementation

This PR tried to better align the two by:

- Use `extractEditorsDropData` to add the uriList
- In the tree view, use `toVSDataTransfer` to convert the event to a `VSDataTransfer`
- Remove `convertKnownMimes` as `extractEditorsDropData` handles this now

### CODE REVIEW

File: src/vs/workbench/browser/parts/views/treeView.ts [hunk 1]
```diff
@@ -1442,9 +1440,9 @@ export class CustomTreeViewDragAndDrop implements ITreeDragAndDrop<ITreeItem> {
...
 	onDragOver(data: IDragAndDropData, targetElement: ITreeItem, targetIndex: number, originalEvent: DragEvent): boolean | ITreeDragOverReaction {
 		const types: Set<string> = new Set();
-		originalEvent.dataTransfer?.types.forEach((value, index) => {
-			if (INTERNAL_MIME_TYPES.indexOf(value) < 0) {
-				types.add(this.convertKnownMimes(value).type);
+		originalEvent.dataTransfer?.types.forEach((type) => {
+			if (INTERNAL_DND_MIME_TYPES.indexOf(type) < 0) {
+				types.add(type === DataTransfers.RESOURCES.toLowerCase() ? Mimes.uriList : type);
 			}
 		});
```
Comment (alexr00) [inline src/vs/workbench/browser/parts/views/treeView.ts:1448]: Something in this change is adding extra mime types that shouldn't be added. The following line logs the mime types. I tested with the tree view sample by dragging one of the tree items onto another. In insiders, the following mime types are listed:
```
TreeView dragged mime types: text/plain, text/uri-list, application/vnd.code.tree.testviewdraganddrop
```
With this change:
```
TreeView dragged mime types: text/plain, text/uri-list, codefiles, application/vnd.code.tree.testviewdraganddrop, codeeditors
```

`codefiles` and `codeeditors` should not appear in this list.

File: src/vs/workbench/browser/parts/views/treeView.ts [hunk 2]
```diff
@@ -1442,9 +1440,9 @@ export class CustomTreeViewDragAndDrop implements ITreeDragAndDrop<ITreeItem> {
 
 	onDragOver(data: IDragAndDropData, targetElement: ITreeItem, targetIndex: number, originalEvent: DragEvent): boolean | ITreeDragOverReaction {
 		const types: Set<string> = new Set();
-		originalEvent.dataTransfer?.types.forEach((value, index) => {
-			if (INTERNAL_MIME_TYPES.indexOf(value) < 0) {
-				types.add(this.convertKnownMimes(value).type);
+		originalEvent.dataTransfer?.types.forEach((type) => {
+			if (INTERNAL_DND_MIME_TYPES.indexOf(type) < 0) {
+				types.add(type === DataTransfers.RESOURCES.toLowerCase() ? Mimes.uriList : type);
```
Comment (alexr00) [inline src/vs/workbench/browser/parts/views/treeView.ts:1445]: This also doesn't properly convert `Files` to `text/uri-list`. `Files` is the mime type used when files are dragged from the OS.

**Review by alexr00 — CHANGES_REQUESTED** [review]
I saw some incorrect behavior while testing, please see the comments!

File: src/vs/workbench/browser/parts/views/treeView.ts (hunk 1 above)
Comment (mjbvz) [reply src/vs/workbench/browser/parts/views/treeView.ts:1448]: Good catch as I believe this is also an issue with the drop into editor and copy paste api. Will see if I can fix this in shared code

**Comment by mjbvz** [comment]
@alexr00 Can you give this another look/test?  I think it's doing the correct thing now but the area is complicated :) 

Some things I learned along the way:

- `Files` is not a real type but is added when the data transfer contains any files (https://www.w3.org/TR/2011/WD-html5-20110113/dnd.html#datatransfer)
- You can't access dragged files until the drop happens (this includes access to the dragged file name). However! the files still exist as items on the data transfer during drag. They even have `kind === 'file'` but `getAsFile()` returns null (our code actually doesn't handle this correctly today but I'll look into this in a separate PR)
- We implement `text/uri-list` incorrect. We need to use `\r\n` as the separator instead of just `\n`. Again I'll look into fixing this in a operate PR instead of making this one even larger

File: src/vs/editor/browser/dnd.ts [hunk 3]
```diff
@@ -31,3 +36,34 @@ export function createFileDataTransferItemFromFile(file: File): IDataTransferIte
...
 	});
 }
+
+const INTERNAL_DND_MIME_TYPES = Object.freeze([
+	CodeDataTransfers.EDITORS,
+	CodeDataTransfers.FILES,
+	DataTransfers.RESOURCES,
+]);
+
+export function addExternalEditorsDropData(dataTransfer: VSDataTransfer, dragEvent: DragEvent) {
```
Comment (alexr00) [inline src/vs/editor/browser/dnd.ts:46]: When dragging a file from the OS into a tree view I now see 2 of the same file getting added by `addExternalEditorsDropData`.

**Review by alexr00 — CHANGES_REQUESTED** [review]
I'm still seeing 1 more behavior difference, but otherwise it's looking good.

**Review by alexr00 — APPROVED** [review]
🚀

**Pull request merged**